client/src/ingredients/examples.js

```diff
@@ -0,0 +1,88 @@
+import React from "react";
+
+function RenderSources({ sources }) {
+  return (
+    <>
+      {sources.html && <h4>HTML</h4>}
+      {/* <code> inside a <pre>?? Is that necessary? */}
+      {sources.html && (
+        <pre>
+          <code>{sources.html}</code>
+        </pre>
+      )}
+
+      {sources.css && <h4>CSS</h4>}
+      {sources.css && (
+        <pre>
+          <code>{sources.css}</code>
+        </pre>
+      )}
+
+      {sources.js && <h4>JavaScript</h4>}
+      {sources.js && (
+        <pre>
+          <code>{sources.js}</code>
+        </pre>
+      )}
+    </>
+  );
+}
+
+function RenderLiveSample({ example }) {
+  const srcdoc = `<html>
+  <head>
+      <meta charset="utf-8">
+      <style type="text/css">${example.sources.css}</style>
+      <title>${example.description.title}</title>
+  </head>
+  <body>${example.sources.html}
+      <script>${example.sources.js || ""}</script>
+  </body>
+</html>`;
+  return (
+    <>
+      <h4>Result</h4>
+      <iframe
+        className="live-sample-frame sample-code-frame"
+        srcDoc={srcdoc}
+        title={example.description.title}
+        id="frame_Live_example"
+        width={example.description.width}
+        height={example.description.height}
+        frameBorder={0}
+      >
+        >
+      </iframe>
+    </>
+  );
+}
+
+function RenderExample({ example }) {
+  return (
+    <>
+      {example.description.title && <h3>{example.description.title}</h3>}
+
+      {example.description.title && (
+        <div
+          dangerouslySetInnerHTML={{ __html: example.description.content }}
+        />
+      )}
+
+      <RenderSources sources={example.sources} />
+
+      {/* XXX this is an odd condition */}
+      {example.description.width && <RenderLiveSample example={example} />}
```

Comment:
It just seems odd to check if a  description has a width to decide whether or not to render the `RenderLiveSample` component. If there's a good reason, can we write a better code comment that justifies the *why*?

Reply:
I will write a comment.

At the moment the presence of width and height indicates that the example should be live sample with an output iframe. I'm not terribly happy about this either and would be happy to consider an extension to the examples structure enabling authors to indicate explicitly that an example is live. But I'd like to address that in a follow-up. I can add a comment to that effect though.

Reply:
-> https://github.com/mdn/stumptown-experiment/issues/45